apps/cms.c

```diff
@@ -51,6 +52,18 @@ static int cms_set_pkey_param(EVP_PKEY_CTX *pctx,
 # define SMIME_ENCRYPTED_ENCRYPT (14 | SMIME_OP)
 # define SMIME_SIGN_RECEIPT      (15 | SMIME_IP | SMIME_OP)
 # define SMIME_VERIFY_RECEIPT    (16 | SMIME_IP)
+# define CMS_CADES                   0x100000
+
+
+struct ESS_cert_id_v2_st {
```

Comment:
Declaring again these structures means that a few access methods are missing.
You better declare them in a first commit.